src/makeCalculator.js

```diff
@@ -37,7 +37,26 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(operation, number) {
+      this.result = operation(this.result, number);
+
+      return this;
+    },
+
+    add: (x, y) => x + y,
+    subtract: (x, y) => x - y,
+    multiply: (x, y) => x * y,
+    divide: (x, y) => x / y,
```

Comment:
what happens when y === 0?

Reply:
.Yes, dividing by 0 will cause an error in the code, so you need to check for such cases. It seems that the rule correctly alerted in the presence of this operation and this number.